### PR TITLE
Hide follower-only birthday reminders

### LINKED ISSUE #6135: UX Annoy: Birthday reminders showing for followers
### Expected behavior

Birthday reminders show only for people I chose to follow.

### Actual behavior

Birthday reminders show also for people who are following me.

screen shot 2018-11-14 at 14 56 53
screen shot 2018-11-14 at 14 57 56

### Friendica version you encountered the problem

2018.12-dev

### PR DESCRIPTION
**Pull request by MrPetovan**
Fixes #6135

I honestly didn't expect to put the foot on such a landmine, but the birthday reminder system truly is inscrutable to me. I briefly planned to rework the whole thing, but there are ramifications related to the birthday field I'm not fully grasping. So I left the confusing logic in place, refactoring the minimum and just adding the required SQL condition for the unexpected birthday reminders to not show up and not being created in the future.

### CODE REVIEW

**Comment by annando** [comment]
I haven't got a chance looking deeply in the code. I can remember that I kept the date constants that way since it hadn't been date fields but text fields. And yes, I don't understand the logic as well and kept my hands off from it. :grin:

**Comment by AndyHee** [comment]
> the birthday reminder system truly is inscrutable

Yes, to put it politely! 

Today is the 23 Nov! The good news is that from tomorrow onwards, we're steadily moving closer to the 23 May.  

screenshot_2018-11-23_21-04-43

File: src/Model/Contact.php [hunk 1]
```diff
@@ -1957,44 +1957,33 @@ public static function removeSharer($importer, $contact, array $datarray = [], $
...
-				q("INSERT INTO `event` (`uid`,`cid`,`created`,`edited`,`start`,`finish`,`summary`,`desc`,`type`,`adjust`)
-				VALUES ( %d, %d, '%s', '%s', '%s', '%s', '%s', '%s', '%s', '%d' ) ", intval($rr['uid']), intval($rr['id']),
-					DBA::escape(DateTimeFormat::utcNow()), DBA::escape(DateTimeFormat::utcNow()), DBA::escape(DateTimeFormat::utc($nextbd)),
-					DBA::escape(DateTimeFormat::utc($nextbd . ' + 1 day ')), DBA::escape($bdtext), DBA::escape($bdtext2), DBA::escape('birthday'),
-					intval(0)
-				);
-
+		$condition = [
+			'`bd` != ""
+			AND `bd` > "0001-01-01"
```
Comment (annando) [inline src/Model/Contact.php]: Shouldn't you use the constant here as well?

**Comment by annando** [comment]
Currently we are storing the birthday in some string a we do this constantly calculation stuff with "bd" and "bdyear" and so on. Just storing the birthday in some date field and doing some date magic to fetch all upcoming birthday events should be enough, shouldn't it?

**Comment by MrPetovan** [comment]
The things I don't understand:
- `bdyear` is updated when a birthday event is created, but `bd` is updated as well with the current year.
- Birthday events creation is triggered for contacts where `bdyear` is different from the extracted year of `bd`
- Consequently, I don't see how further birthday events are created beyond the first.
- Updating `bd` means we don't keep the birth year information if it was provided, so we can't show age for exampe.

**Comment by annando** [comment]
Yes, this sums it up.

**Pull request merged**